bitsandbytes/autograd/_functions.py

```diff
@@ -318,13 +324,14 @@ def forward(ctx, A, B, out=None, bias=None, state=MatmulLtState):
             state.outlier_pool = GlobalOutlierPooler.get_instance()
 
         # Cast A to fp16
-        if A.dtype != torch.float16:
-            warnings.warn(f"MatMul8bitLt: inputs will be cast from {A.dtype} to float16 during quantization")
+        ctx.cast_dtype = torch.float16
+        if A.dtype != ctx.cast_dtype:
```

Comment:
Is this a necessary change in this PR? The original code hard-coded float16. It should be ok for CUDA?

Reply:
yes, should be fine for CUDA. cast_type intends to leave the choice for more dtypes but not actually used in this PR, here have removed it from this PR.